projects/angular-split/src/lib/component/split.component.ts

```diff
@@ -104,12 +114,33 @@ import { ANGULAR_SPLIT_DEFAULT_OPTIONS } from '../angular-split-config.token'
         [attr.aria-valuenow]="area.size"
         [attr.aria-valuetext]="getAriaAreaSizeText(area.size)"
       >
-        <div class="as-split-gutter-icon"></div>
+        <ng-container *ngIf="customGutter?.template; else defaultGutterTpl">
+          <ng-container *asSplitGutterDynamicInjector="index + 1; let injector">
+            <ng-container
+              *ngTemplateOutlet="
+                customGutter.template;
+                context: {
+                  areaBefore: area,
+                  areaAfter: displayedAreas[index + 1],
+                  gutterNum: index + 1,
+                  first,
+                  last: index === displayedAreas.length - 2,
+                  isDragged: draggedGutterNum === index + 1
+                };
+                injector: injector
+              "
+            ></ng-container>
+          </ng-container>
+        </ng-container>
+        <ng-template #defaultGutterTpl>
+          <div class="as-split-gutter-icon"></div>
+        </ng-template>
       </div>
     </ng-template>`,
   encapsulation: ViewEncapsulation.Emulated,
 })
 export class SplitComponent implements AfterViewInit, OnDestroy {
+  @ContentChild(SplitGutterDirective) customGutter: SplitGutterDirective
```

Comment:
This currently allows full customization but for a specific `as-split` instance. If someone wants a global custom split for the whole app it is currently not supported.
I thought maybe we should add to `IDefaultOptions` a new option called gutter component which will be the new default gutter. The component will have access to `GUTTER_NUM_TOKEN` which will probably need to change to `GUTTER_CONTEXT_TOKEN` with same data as the template context.
What do you think @beeman?

Another option is leaving this to later using theme customization. Will be more restricted though than a component.

Reply:
I think it's probably good to leave this for later when work on the theme is done.